### DISCUSSION TITLE
Inquiry Regarding MTEX's Capability to Recognize Coherent and Incoherent Twin Boundaries

### DISCUSSION THREAD

**Discussion by PengWei97** [mtex-toolbox/mtex#2136] [Feature Requests]
Dear MTEX Support Team,

I'm reaching out to inquire about MTEX's capability to identify coherent and incoherent twin boundaries in crystallographic microstructures.

I've come across an algorithm mentioned in a publication available at https://pubmed.ncbi.nlm.nih.gov/11996188/, which addresses twin boundary identification. I'm interested to know if MTEX provides similar functionalities.

Could you please confirm if MTEX offers features for recognizing coherent and incoherent twin boundaries? If so, any guidance on how to utilize these functionalities effectively would be greatly appreciated.

Thank you for your assistance.

Best regards,

Wei

**Comment by kilir** [comment]
Hi,
did you have a look here https://mtex-toolbox.github.io/TwinningBoundaries.html ?
Maybe that helps already.

Cheers,

Rüdiger

**Reply by PengWei97** [reply]
Dear MTEX expert,

Thank you for reaching out. I wanted to inquire whether the twinning identification method provided by MTEX is equivalent to the criteria outlined in paper: [Extracting twins from orientation imaging microscopy scan data - PubMed (nih.gov)](https://pubmed.ncbi.nlm.nih.gov/11996188/) .

In this paper, it is stated that, " Two criteria must be satisfied before a given grain boundary can be strictly considered a twin boundary. First, the misorientation between the crystal lattices separated by the boundary must satisfy a specific misorientation relationship.  The second criterion is that the boundary plane separating the two grains must be coincident with a particular twinning plane."

MTEX's Twinning Analysis utilizes the following command for identification:

```bash
isTwinning = angle(gB_MgMg.misorientation, twinning) < 5*degree;
```

From my understanding, MTEX identifies twinning when the difference between the rotation matrices of neighboring grains (gB_MgMg.misorientation) and the rotation matrix from the twin to the matrix grain is less than a specified tolerance.

Therefore, my question is whether my understanding of MTEX's twinning identification is correct, and whether the criteria mentioned in this paper and those implemented in MTEX are equivalent?

Thank you for your assistance and clarification.

Sincerely,
Wei

**Reply by kilir** [reply]
Correct, in the documentation only the misorientation is considered. While I'd say the second criterion does not apply for all twins (there are some which can have quite arbitrary contact planes), you could consider the trace of the potential contact plane as a second  condition, unless you have data which gives you the 3D plane normals as well.
Cheers,
Rüdiger

**Reply by PengWei97** [reply]
Yes, the paper also mentions that EBSD data only contain information about the orientation of planes. Therefore, their proposed strategy involves comparing the misorientation between the trace of grain boundaries and twin planes. The paper claims a high accuracy rate of 90% compared to three-dimensional studies.

Additionally, I found a post on the MTEX forum regarding the identification of coherent and incoherent twin boundaries, [link to post](https://groups.google.com/g/mtexmail/c/aZ9GBZCiXxQ/m/UjVpJZs1AAAJ). The provided MATLAB code distinguishes between coherent and incoherent twin boundaries based on the trend of the boundary direction close to a  twinning plane, 
```matlab
% select CSL(3) grain boundaries
gB3 = gB_FeFe(angle(gB_FeFe.misorientation,CSL(3,ebsd('Iron fcc').CS)) < 8.66*degree);
% gb direction in crystal coordinates
gbdirc = inv(grains(gB3.grainId).meanOrientation).*gB3.direction;
% angle between gbdirc and {1,1,1}
MAngle = angle(gbdirc,Miller(1,1,1,ebsd('Iron fcc').CS));
% select only boundaries which have a trend very close to {1,1,1}
GBcoherent = gB3(any(MAngle < 5*degree,2));
% overlay grain boundaries with the existing plot
hold on
plot(gB3,'lineColor','r','linewidth',1.0,'DisplayName','CSL 3');
hold on
plot(GBcoherent,'lineColor','g','linewidth',1.5,'DisplayName','Coherent');
hold off
```

Currently, I haven't implemented this code for identifying HCP-Ti. However, it should involve calculating the orientation relationship between grain boundaries and twin planes after identifying twins. I'm not sure if my understanding is correct or if this approach is feasible.

Best regards,
Wei
